adcm_aio_client/core/objects/cm.py

```diff
@@ -41,7 +41,27 @@ def get_own_path(self: Self) -> Endpoint:
         return ("adcm",)
 
 
-class License(InteractiveObject): ...
+class License(WithProtectedRequester):
+    def __init__(self: Self, requester: Requester, prototypes_data: dict) -> None:
+        self._license_prototype_id = prototypes_data["id"]
+        self._data = prototypes_data["license"]
+        self._requester = requester
+
+    @property
+    def text(self: Self) -> str:
+        return str(self._data["text"])
+
+    @property
+    def state(self: Self) -> Literal["absent", "accepted", "unaccepted"]:
+        return self._data["status"]
+
+    async def accept(self: Self) -> str:
+        accept_response = await self._requester.post(
+            "prototypes", self._license_prototype_id, "license", "accept", data={}
+        )
+        if accept_response.response.status_code != 200:  # type: ignore
+            self._data["status"] = "accepted"
```

Comment:
what will happen if there's no license?

if it'll be "accepted", then it's status still be absent I believe

are there difference in response status codes? why check for 200 exactly?

unsuccessful requests will raise error anyway

Reply:
if there is no license there won't be status code 200. There would be status code 409 and the condition would fail. I don't see the problem